omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_0/PasswordResource2_0.java

```diff
@@ -0,0 +1,98 @@
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+import org.openmrs.module.webservices.rest.web.v1_0.wrapper.openmrs1_8.UserAndPassword1_8;
+import org.openmrs.util.PrivilegeConstants;
+
+@Resource(name = RestConstants.VERSION_1 + "/password", supportedClass = UserAndPassword1_8.class, supportedOpenmrsVersions = { "2.0.*,2.1.*" })
+public class PasswordResource2_0 extends DelegatingCrudResource<UserAndPassword1_8> {
+	
+	@Override
+	public UserAndPassword1_8 newDelegate() {
+		return new UserAndPassword1_8();
+	}
+	
+	@Override
+	public Object create(SimpleObject propertiesToCreate, RequestContext context) throws ResponseException {
+		User authenticatedUser = Context.getAuthenticatedUser();
+		if (authenticatedUser != null) {
+			return update(authenticatedUser.getUuid(), propertiesToCreate, context);
+		} else {
+			throw new APIAuthenticationException("You are not authenticated to change password");
+		}
+	}
+	
+	@Override
+	public UserAndPassword1_8 save(UserAndPassword1_8 user) {
+		User openmrsUser = user.getUser();
+		if (openmrsUser == null || openmrsUser.getUserId() == null) {
+			throw new APIAuthenticationException("User not available");
+		} else {
+			//only authenticated users can change their own password or other's password
+			User authenticatedUser = Context.getAuthenticatedUser();
+			if (authenticatedUser.hasPrivilege(PrivilegeConstants.EDIT_USER_PASSWORDS)) {
+				Context.getUserService().changePassword(openmrsUser, user.getPassword());
+			} else if (authenticatedUser.equals(openmrsUser)) {
+				Context.getUserService().changePassword(user.getOldPassword(), user.getPassword());
+			} else {
+				throw new APIAuthenticationException("You doesn't have privilege to change others password");
+			}
+		}
+		return new UserAndPassword1_8(openmrsUser);
+	}
+	
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addRequiredProperty("password");
+		description.addProperty("oldPassword");
+		
+		return description;
+	}
+	
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		return null;
+	}
+	
+	@Override
+	public UserAndPassword1_8 getByUniqueId(String uuid) {
+		try {
+			Context.addProxyPrivilege(PrivilegeConstants.GET_USERS);
+			return new UserAndPassword1_8(Context.getUserService().getUserByUuid(uuid));
+		}
+		finally {
+			Context.removeProxyPrivilege(PrivilegeConstants.GET_USERS);
+		}
+	}
+	
+	@Override
+	protected void delete(UserAndPassword1_8 delegate, String reason, RequestContext context) throws ResponseException {
+		throw new ResourceDoesNotSupportOperationException("This operation is not supported");
+	}
+	
+	@Override
+	public void purge(UserAndPassword1_8 delegate, RequestContext context) throws ResponseException {
+		throw new ResourceDoesNotSupportOperationException("This operation is not supported");
+	}
+}
```

Comment:
I just realized we should also override this, to also throw ResourceDoesNotSupportOperationException:

	public Object retrieve(String uuid, RequestContext context) throws ResponseException